src/core/dep-graph.ts

```diff
@@ -138,6 +138,11 @@ class DepGraphImpl implements types.DepGraphInternal {
     return count;
   }
 
+  public equals(other: types.DepGraph, compareRoot: boolean = false): boolean {
```

Comment:
would suggest 2 small changes:
* put the `compareRoot` into an options object, this way it will be easier to read & reason about the code that calls this.
* make `compareRoot` true by default, as the root package today is more part of the graph than isn't. e.g the `getPkgs()` returns the root too.